app/models/media.rb

```diff
@@ -77,25 +79,36 @@ def domain
     ''
   end
 
-  def self.find_or_create_from_original_claim(claim, project_media_team)
-    if claim.match?(/\A#{URI::DEFAULT_PARSER.make_regexp(['http', 'https'])}\z/)
-      uri = URI.parse(claim)
-      content_type = Net::HTTP.get_response(uri)['content-type']
-      ext = File.extname(uri.path)
-
-      case content_type
-      when /^image\//
-        find_or_create_uploaded_file_media_from_original_claim('UploadedImage', claim, ext)
-      when /^video\//
-        find_or_create_uploaded_file_media_from_original_claim('UploadedVideo', claim, ext)
-      when /^audio\//
-        find_or_create_uploaded_file_media_from_original_claim('UploadedAudio', claim, ext)
-      else
-        find_or_create_link_media_from_original_claim(claim, project_media_team)
+  def self.find_or_create_media_associated_to(project_media)
+    m = nil
+    original_claim = project_media.set_original_claim&.strip
+    media_type = project_media.media_type
+    team = project_media.team
+
+    if original_claim
+      case media_type
+      when 'UploadedImage', 'UploadedVideo', 'UploadedAudio'
+        m = find_or_create_uploaded_file_media(original_claim, media_type, true)
+      when 'Claim'
+        m = find_or_create_claim_media(original_claim, nil, true)
+      when 'Link'
+        m = find_or_create_link_media(original_claim, team, true)
+      when 'Blank'
+        m = Blank.create!
       end
     else
-      find_or_create_claim_media_from_original_claim(claim)
+      case media_type
+      when 'UploadedImage', 'UploadedVideo', 'UploadedAudio'
+        m = find_or_create_uploaded_file_media(project_media.file, media_type)
+      when 'Claim'
+        m = find_or_create_claim_media(project_media.quote, project_media.quote_attributions)
+      when 'Link'
+        m = find_or_create_link_media(project_media.url, team)
+      when 'Blank'
+        m = Blank.create!
+      end
```

Comment:
Let's try to resolve this duplication by trying to send the media information as an argument. Suggested signature:

```ruby
find_or_create_media_from_content(type, content, additional_args = [])
```